### ISSUE TITLE
MaskZeroCriterion:recursiveMask fails when mask is empty

### ISSUE THREAD

**Issue by notimesea** [Element-Research/rnn#140]
My model and criterion are as follows:
```lua
model = nn.Sequential()
model:add(nn.Convert(ds:ioShapes(), 'bwh'))
model:add(nn.SplitTable(1, 2))

model:add(
  nn.Sequencer(
    nn.MaskZero(
      nn.Sequential()
        :add(nn.FastLSTM(inputSize, hiddenSize))
        :add(nn.Dropout(0.2))
        :add(nn.Linear(hiddenSize, outputSize))
        :add(nn.LogSoftMax())
    , 1)
  )
)

criterion = nn.ModuleCriterion( 
  nn.SequencerCriterion(
    nn.MaskZeroCriterion(
      nn.ClassNLLCriterion(), 1
    )
  ),
  nn.Identity(),
  nn.Sequential()
    :add(nn.Convert())
    :add(nn.SplitTable(1, 1))
)
```

My dataset is like:
```lua
local samples = 10
local maxLen = 20
inputs = torch.Tensor(samples, maxLen, inputSize)
targets = torch.LongTensor(samples, maxLen)
```
Inputs are padded by zeros in the very beginning of each sequence.

When learning error occurs:
```
/home/evgeny/torch/install/bin/luajit: ...ny/torch/install/share/lua/5.1/rnn/MaskZeroCriterion.lua:34: bad argument #1 to 'index' (sizes do not match at /home/evgeny/torch/extra/cutorch/lib/THC/generic/THCTensorCopy.c:47)
stack traceback:
	[C]: in function 'index'
	...ny/torch/install/share/lua/5.1/rnn/MaskZeroCriterion.lua:34: in function 'recursiveMask'
	...ny/torch/install/share/lua/5.1/rnn/MaskZeroCriterion.lua:76: in function 'forward'
	...y/torch/install/share/lua/5.1/rnn/SequencerCriterion.lua:39: in function 'forward'
	...eny/torch/install/share/lua/5.1/dpnn/ModuleCriterion.lua:23: in function 'forward'
	...torch/install/share/lua/5.1/dp/propagator/propagator.lua:168: in function 'forward'
	.../torch/install/share/lua/5.1/dp/propagator/optimizer.lua:50: in function 'propagateBatch'
	...torch/install/share/lua/5.1/dp/propagator/propagator.lua:127: in function 'propagateEpoch'
	...torch/install/share/lua/5.1/dp/propagator/experiment.lua:110: in function 'run'
	conf_rnn.lua:246: in main chunk
	[C]: in function 'dofile'
	...geny/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
	[C]: at 0x00406670
```

As far as I understood this error caused by empty mask as every sequence in batch is started with zeros.

My questions are:
Is my model or criterion used in right way?
How can I avoid empty mask using MaskZeroCriterion?

**Comment by nicholas-leonard** [comment]
@notimesea This should be fixed by https://github.com/Element-Research/rnn/commit/4c4895a2da501ac2b6c66eec4d174ca12c3f5a1d . Let me know if this works for you. Thanks!
